lang/java/reef-runtime-multi/src/main/java/org/apache/reef/runtime/multi/driver/RuntimesHost.java

```diff
@@ -0,0 +1,171 @@
+        //it to instantiate the actual runtime.
+        Injector rootInjector = Tang.Factory.getTang().newInjector();
+        initializeInjector(rootInjector);
+        final JavaConfigurationBuilder cb = Tang.Factory.getTang().newConfigurationBuilder();
+        cb.bindNamedParameter(RuntimeName.class, rd.getRuntimeName().toString());
+        cb.bindImplementation(Runtime.class, RuntimeImpl.class);
+
+        AvroConfigurationSerializer serializer = new AvroConfigurationSerializer();
+        Configuration config = serializer.fromString(rd.getSerializedConfiguration().toString());
+        final Injector runtimeInjector = rootInjector.forkInjector(config, cb.build());
+        this.runtimes.put(rd.getRuntimeName().toString(), runtimeInjector.getInstance(Runtime.class));
+      } catch (InjectionException e) {
+        throw new RuntimeException("Unable to initialize runtimes.", e);
+      } catch (IOException e) {
+        throw new RuntimeException("Unable to initialize runtimes.", e);
+      }
+    }
+  }
+
+  /**
+   * Initializes injector by copying needed handlers.
+   * @param runtimeInjector Teh injector to initialize
+   * @throws InjectionException
+   */
+  private void initializeInjector(final Injector runtimeInjector) throws InjectionException {
+    final EventHandler<ResourceStatusEvent> statusEventHandler =
+            this.originalInjector.getNamedInstance(RuntimeParameters.ResourceStatusHandler.class);
+    runtimeInjector.bindVolatileParameter(RuntimeParameters.ResourceStatusHandler.class, statusEventHandler);
+    final EventHandler<NodeDescriptorEvent> nodeDescriptorEventHandler =
+            this.originalInjector.getNamedInstance(RuntimeParameters.NodeDescriptorHandler.class);
+    runtimeInjector.bindVolatileParameter(RuntimeParameters.NodeDescriptorHandler.class, nodeDescriptorEventHandler);
+    final EventHandler<ResourceAllocationEvent> resourceAllocationEventHandler =
+            this.originalInjector.getNamedInstance(RuntimeParameters.ResourceAllocationHandler.class);
+    runtimeInjector.bindVolatileParameter(
+            RuntimeParameters.ResourceAllocationHandler.class,
+            resourceAllocationEventHandler);
+    final EventHandler<RuntimeStatusEvent> runtimeStatusEventHandler =
+            this.originalInjector.getNamedInstance(RuntimeParameters.RuntimeStatusHandler.class);
+    runtimeInjector.bindVolatileParameter(
+            RuntimeParameters.RuntimeStatusHandler.class,
+            runtimeStatusEventHandler);
+  }
+
+  /**
+   * Retrieves requested runtime, if requested name is empty a default runtime will be used.
+   * @param requestedRuntimeName teh requested runtime name
+   * @return
+   */
+  private Runtime getRuntime(final String requestedRuntimeName) {
+    String runtimeName = requestedRuntimeName;
+    if (StringUtils.isEmpty(runtimeName) || StringUtils.isBlank(runtimeName)) {
```

Comment:
[`isBlank`](http://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/StringUtils.html#isBlank%28java.lang.String%29) implies `isEmpty`. This can be rewritten as

```java
if (StringUtils.isBlank(runtimeName)) {
```